src/app/store/selectors/VehicleTechRecordModel.selectors.ts

```diff
@@ -12,22 +18,69 @@ export const selectVehicleTechRecordModelHavingStatusAll = createSelector(
   (state: IVehicleTechRecordModelState) => state.vehicleTechRecordModel
 );
 
-export const selectSelectedVehicleTechRecordModel = createSelector(
-  selectFeature,
-  (state: IVehicleTechRecordModelState) => state.selectedVehicleTechRecordModel
-);
-
 export const getSelectedVehicleTechRecord = createSelector(
   selectFeature,
   (state: IVehicleTechRecordModelState) => state.selectedVehicleTechRecord
 );
 
+export const getActiveVehicleTechRecord = createSelector(
+  // getUserDetails,
+  getSelectedVehicleTechRecord,
+  (vehicleTechRecord) => (searchRecord: TechRecord = null): VehicleTechRecordEdit => {
+    const primaryVrm: VrmModel = vehicleTechRecord.vrms.find((vrm) => vrm.isPrimary);
+
+    return {
+      msUserDetails: null,
+      vin: vehicleTechRecord.vin,
+      primaryVrm: primaryVrm ? primaryVrm.vrm : '',
+      secondaryVrms: vehicleTechRecord.vrms
+        .filter((vrm) => !vrm.isPrimary)
+        .map((secondaryVrm) => secondaryVrm.vrm),
+      // trailerId:
+      techRecord: [getTechRecord(vehicleTechRecord.techRecord, searchRecord)]
+    };
+  }
+);
+
 export const getVehicleTechRecordMetaData = createSelector(
   getSelectedVehicleTechRecord,
-  (state: VehicleTechRecordModel) => state.metadata
+  (vRecord: VehicleTechRecordModel) => vRecord.metadata
 );
 
-export const getViewState = createSelector(
+export const getTechViewState = createSelector(
   selectFeature,
   (state: IVehicleTechRecordModelState) => state.viewState
 );
+

```

Comment:
suggest to add a comment here to indicate Util functions within `VehicleTechRecordModel.selectors`